pallets/services/src/functions/evm_hooks.rs

```diff
@@ -828,10 +790,51 @@ impl<T: Config> Pallet<T> {
 		)
 	}
 
+	/// Hook to be called upon a new RPC address update on a blueprint.
+	///
+	/// This function is called when the RPC address is updated. It performs an EVM call
+	/// to the `onUpdateRpcAddress` function of the service blueprint's manager contract.
+	///
+	/// # Parameters
+	/// * `blueprint` - The service blueprint.
+	/// * `blueprint_id` - The blueprint ID.
+	/// * `preferences` - The operator preferences.
+	///
+	/// # Returns
+	///
+	/// * `Result<(bool, Weight), DispatchErrorWithPostInfo>` - A tuple containing a boolean
+	///   indicating whether the RPC address update is allowed and the weight of the operation.
+	pub fn on_update_rpc_address_hook(
+		blueprint: &ServiceBlueprint<T::Constraints>,
+		blueprint_id: u64,
+		preferences: &OperatorPreferences<T::Constraints>,
+	) -> Result<(bool, Weight), DispatchErrorWithPostInfo> {
+		#[allow(deprecated)]
+		Self::dispatch_hook(
+			blueprint,
+			Function {
+				name: String::from("onUpdateRpcAddress"),
+				inputs: vec![
+					ethabi::Param {
+						name: String::from("blueprintId"),
+						kind: ethabi::ParamType::Uint(64),
+						internal_type: None,
+					},
+					OperatorPreferences::<T::Constraints>::to_ethabi_param(),
+				],
+				outputs: Default::default(),
+				constant: None,
+				state_mutability: StateMutability::Payable,
+			},
+			&[Token::Uint(ethabi::Uint::from(blueprint_id)), preferences.to_ethabi()],
+			Zero::zero(),
+		)
```

Comment:
You need to update the MasterServiceBlueprint contract and the base contract here: https://github.com/tangle-network/tnt-core/tree/main/src and update the version in the foundry.toml